lib/wallet/txdb.js

```diff
@@ -1777,17 +1778,17 @@ class TXDB {
       case types.REGISTER: {
         assert(i < tx.inputs.length);
 
-        const nameHash = covenant.getHash(0);
-        const prevout = tx.inputs[i].prevout;
+        const coins = await this.getSpentCoins(tx);
```

Comment:
I don't this is covered by the test yet. You could add another `it` block at the end of the test you already have, store the latest block entry in a local variable, call `walletDB.removeBlock()`, then assert the balances are all as expected.